src/sysext/sysext.c

```diff
@@ -2208,6 +2209,53 @@ static int vl_method_refresh(sd_varlink *link, sd_json_variant *parameters, sd_v
         return sd_varlink_reply(link, NULL);
 }
 
+static int parse_version(const char *filename, char **ret) {
+        const char *start, *last_dot_after_underscore;
+        char *version = NULL;
+
+        assert(filename);
+        assert(ret);
+
+        /* Find the last occurrences of '_'. This will mark the start of the extension version. */
+        start = strrchr(filename, '_');
+        if (!start) {
+                /* If no underscore found, assign "none" and return success.
+                * This is for cases where there are no versions in the filename like 'image.raw'. */
+                version = strdup("none");
+                goto finish;
+        }
+        start ++;
+
+        last_dot_after_underscore = strrchr(start, '.');
+        if (!last_dot_after_underscore) {
+                /* If no dot found, treat everything after '_' as the version */
+                version = strdup(start);
+                goto finish;
+        }
+
+        if (last_dot_after_underscore[1] && isdigit(last_dot_after_underscore[1])) {
+                /* If the last dot is followed by a digit, the version is everything after the underscore
+                * This is for cases where the extension name is like so: image_0.1 */
+                version = strdup(start);
+        } else if (last_dot_after_underscore[1] && isalpha(last_dot_after_underscore[1])) {
```

Comment:
you can use `char *m = ENDSWITH_SET(filename, ".sysext.raw", ".confext.raw", ".raw")`

Reply:
That would be the same number of if statements, just a few more in verb_list and a few less in parse_version right? Because you would still have to construct the string after checking 'if' the file name has a .raw/ equivalent suffix and has an underscore and so on?